CMakeLists.txt

```diff
@@ -1,5 +1,15 @@
 cmake_minimum_required(VERSION 3.15)
 
+# Skip generating proto files with Protobuf compiler and use prebuilt files
+# Will still build Protobuf as libprotobuf is required for linking
+option(WSL2_CROSS_COMPILE "Building on WSL2 for Windows" OFF)
+if (WSL2_CROSS_COMPILE AND UNIX AND NOT APPLE)
+    set(CMAKE_TOOLCHAIN_FILE cmake/toolchain/x86-64-w64-mingw32.cmake)
+elseif (WSL2_CROSS_COMPILE AND WIN32)
+    message(FATAL_ERROR "Invalid environment.\
```

Comment:
Well, actually it'll not throw if we have APPLE and WSL2 on. Thats why I used more complex condition before.